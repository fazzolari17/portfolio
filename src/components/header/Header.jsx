import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import style from './styleSheet';

const Header = () => {
  const loc = useLocation();
  const home = '/';
  const aboutMe = '/aboutMe';
  const projects = '/projects';
  const contact = '/contact';



  const addUnderlineToCurrentPage = (location, page) => {
    if (location === page) return style.currentPage;
    return style.link;
  };

  return (
    <header data-testid='header' style={style.header}>
      <Link to={home}><div data-testid={'logoG'} className="logo">G<span data-testid={'logoF'} className="tilt">F</span></div></Link>
      <nav style={style.nav}>
        <Link style={addUnderlineToCurrentPage(loc.pathname, home)} to={home}>Home</Link>
        <Link style={addUnderlineToCurrentPage(loc.pathname, aboutMe)} to={aboutMe}>About Me</Link>
        <Link style={addUnderlineToCurrentPage(loc.pathname, projects)} to={projects}>Projects</Link>
        <Link style={addUnderlineToCurrentPage(loc.pathname, contact)} to={contact}>Contact</Link>
      </nav>
    </header>
  );
};

export default Header;
