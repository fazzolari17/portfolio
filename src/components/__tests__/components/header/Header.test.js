import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../../../header/Header';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Header', () => {
  describe('Desktop Header', () => {
    beforeEach(() => {
      render(
        <Router>
          <Header />
        </Router>
      );
    });

    it('renders correctly', () => {
      const tree = renderer.create(
        <Router>
          <Header />
        </Router>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders and dispays menu', () => {
      const home = screen.getByText(/home/i);
      const aboutMe = screen.getByText(/about me/i);
      const projects = screen.getByText(/projects/i);
      const contact = screen.getByText(/contact/i);
      expect(home).toBeInTheDocument();
      expect(aboutMe).toBeInTheDocument();
      expect(projects).toBeInTheDocument();
      expect(contact).toBeInTheDocument();
    });

    it('renders the logo', () => {
      const logoG = screen.getByTestId('logoG');
      const logoF = screen.getByTestId('logoF');
      expect(logoG).toBeInTheDocument();
      expect(logoF).toBeInTheDocument();
    });

  });
});