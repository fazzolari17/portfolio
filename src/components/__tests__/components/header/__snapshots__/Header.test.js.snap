// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Desktop Header renders correctly 1`] = `
<header
  data-testid="header"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "var(--header-background, #071013)",
      "color": "var(--header-text, #F0F8FF)",
      "display": "flex",
      "fontFamily": "Inter, sans-serif",
      "fontWeight": "500",
      "height": "60px",
      "justifyContent": "space-between",
      "left": 0,
      "listStyle": "none",
      "padding": "1em",
      "position": "fixed",
      "textDecoration": "none",
      "textTransform": "uppercase",
      "top": 0,
      "width": "100vw",
      "zIndex": "10",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="logo"
      data-testid="logoG"
    >
      G
      <span
        className="tilt"
        data-testid="logoF"
      >
        F
      </span>
    </div>
  </a>
  <nav
    style={
      Object {
        "marginRight": "2rem",
      }
    }
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "borderBottom": "2px solid var(--header-text, #F0F8FF)",
          "color": "var(--header-text, #F0F8FF)",
          "listStyle": "none",
          "marginLeft": "2em",
          "textDecoration": "none",
          "textTransform": "uppercase",
        }
      }
    >
      Home
    </a>
    <a
      href="/aboutMe"
      onClick={[Function]}
      style={
        Object {
          "color": "var(--header-text, #F0F8FF)",
          "listStyle": "none",
          "marginLeft": "2em",
          "textDecoration": "none",
          "textTransform": "uppercase",
        }
      }
    >
      About Me
    </a>
    <a
      href="/projects"
      onClick={[Function]}
      style={
        Object {
          "color": "var(--header-text, #F0F8FF)",
          "listStyle": "none",
          "marginLeft": "2em",
          "textDecoration": "none",
          "textTransform": "uppercase",
        }
      }
    >
      Projects
    </a>
    <a
      href="/contact"
      onClick={[Function]}
      style={
        Object {
          "color": "var(--header-text, #F0F8FF)",
          "listStyle": "none",
          "marginLeft": "2em",
          "textDecoration": "none",
          "textTransform": "uppercase",
        }
      }
    >
      Contact
    </a>
  </nav>
</header>
`;
